### DISCUSSION TITLE
Farbpalette

### DISCUSSION THREAD

**Discussion by imperiumsources** [mhoeher/opentodolist#89] [General]
Hallo,

habe heute dein Programm entdeckt und gleich unter GNOME 4 als flatpak installiert. Obwohl es ein QT Programm ist, funktioniert es unter GNOME4 erstaunlich gut.  

ich bin unbedarft an dieses Programm angegangen und arbeite mich ein. Als erstes ist mir die Farbgebung aufgefallen.
Siehe Screenshots.

Ich finde diese Farben sehen blass aus und die Auswahl der Farben ist sehr übersichtlich gestaltet. Kann man das ändern und ist hier eine Verbesserung möglich?

VG Richard
Fedora 38, GNOME 44.2 

![Bildschirmfoto vom 2023-07-09 11-05-22](https://github.com/mhoeher/opentodolist/assets/75203950/b313a6db-3047-41d5-8a89-3f8525825491)
![Bildschirmfoto vom 2023-07-09 10-59-10](https://github.com/mhoeher/opentodolist/assets/75203950/39a6ef49-cc3d-4d0d-9971-1a3735b1baf1)

**Comment by mhoeher** [comment]
Hallo @imperiumsources,

die App und auch die meisten der verwendeten Farben basieren auf Google's Material Design. Heißt auch, dass gerade bei eingestelltem "Dark Theme" (die App sollte auf den meisten Systemen automatisch das Systemthema anwenden) viele der Farben eher blass werden. Hier ein Link in die Qt Hilfe zum Thema: https://doc.qt.io/qt-6/qtquickcontrols-material.html#pre-defined-material-colors

Die Farben der "Kacheln" werden durch das Mischen der Basis-Farbe für den Hintergrund mit der ausgewählten Farbe für die Kachel festgelegt. Das führt gerade auch bei eingestelltem Dark Theme zu eher blassen Farben.

Hier überlege ich schon seit einiger Zeit eine Veränderung. Es gibt (offline, d.h. auch noch nicht als Ticket abgebildet) Anfragen, beliebige Farben für die Karten verwenden zu können. Wenn ein solches Feature kommen würde, würde ich es so implementieren, dass die gewählte Farbe direkt für die Kachel verwendet und die Textfarbe entsprechend angepasst wird.

**Reply by imperiumsources** [reply]
Hallo,

das wäre gut und macht die Anwendung schöner. Vielleicht wäre Theming der ganze App eine Möglichkeit. Das hätte den Vorteil das je nach Desktop (z.B KDE / QT / GNOME ) das aussehen angepasst werden kann. Somit würde es nicht als "Fremdkörper"
aussehen. Erweiterungen zu implementieren kostet viel Zeit. 

VG Richard

**Reply by mhoeher** [reply]
Ich verstehe warum einige User Theming gern hätten, aber ja: Das zu implementieren kostet viel Zeit und vor allem ist es schwer so etwas konsistent zu machen (ich erinnere mich, dass z.B: eine Reihe von GNOME Entwicklern eine Kampagne gestartet hat, die sich dagegen ausspricht, dass einzelne Distros etwa GTK Themes anpassen, was dann einige Apps z.T. unbenutzbar gemacht hat).

Eine verbesserte Farbpalette sollte allerdings einfacher machbar sein!
